src/observable/GenerateObservable.ts

```diff
@@ -0,0 +1,129 @@
+import {Observable} from '../Observable' ;
+import {Scheduler} from '../Scheduler';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+import {Action} from '../scheduler/Action';
+
+import {isScheduler} from '../util/isScheduler';
+
+const selfSelector = <T>(value: T) => value;
+
+export type ConditionFunc<S> = (state: S) => boolean;
+export type IterateFunc<S> = (state: S) => S;
+export type ResultFunc<S, T> = (state: S) => T;
+
+interface SchedulerState<T, S> {
+  state: S;
+  params: {
+    subscriber: Subscriber<T>;
+    condition?: ConditionFunc<S>;
+    iterate: IterateFunc<S>;
+    resultSelector: ResultFunc<S, T>;
+  };
+}
+
+export interface GenerateOptions<T, S> {
+  initialState: S;
+  condition?: ConditionFunc<S>;
+  iterate: IterateFunc<S>;
+  resultSelector: ResultFunc<S, T>;
+  scheduler?: Scheduler;
+}
+
+export class GenerateObservable<T, S> extends Observable<T> {
+  constructor(
+    private initialState: S,
+    private condition: ConditionFunc<S>,
+    private iterate: IterateFunc<S>,
+    private resultSelector: ResultFunc<S, T>,
+    private scheduler?: Scheduler) {
+      super();
+  }
+
+  static create<T, S>(
+    initialState: S,
```

Comment:
it's just style - would you able to flatten signature into single line similar to [`combineLatest`](https://github.com/ReactiveX/rxjs/blob/master/src/operator/combineLatest.ts#L45-L60) ? it's not strict rules but generally follows that way for convenience comparison between signatures.